### PR TITLE
Move and update appengine-endpoints-helloendpoints-java-maven sample

### PR DESCRIPTION
**Pull request by frankyn**
* This is an updated version of the [appengine-endpoints-helloendpoints-java-maven](https://github.com/GoogleCloudPlatform/appengine-endpoints-helloendpoints-java-maven) repository.
* Migrated sample to Endpoints Frameworks v2.0
* Added gradle build support
* Added support for new AppEngine plugin 
* Added comments on how to use Endpoints Frameworks v1.0. It's commented unless developer wants to use this framework version. 
* Updated README.md to reflect changes with new Endpoints Frameworks v2.0 tooling

### CODE REVIEW

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/Jenkins.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/Jenkins.sh]: Wow - what a long path `endpoints-v1-using-endpoints-frameworks-v2`.  Seems quite wordy, but Ok if you don't want to use abbreviations.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/Jenkins.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/Jenkins.sh]: We don't use: All Rights Reserved.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/build.gradle [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/build.gradle]: Strike All Rights Reserved.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/UpdateApplication.launch [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/UpdateApplication.launch]: Shouldn't this be Deploy? -- we may wish to drop this.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/DevAppServer.launch [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/DevAppServer.launch]: We may wish to drop this, if not, it should have a copyright.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/UpdateApplication.launch (hunk 4 above)
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/UpdateApplication.launch]: Copyright  if you keep.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/nbactions.xml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/nbactions.xml]: strike All Rights Reserved.

Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/nbactions.xml]: We may wish to strike this.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml [hunk 7]
```diff
@@ -0,0 +1,179 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<!--
+Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml]: Strike All Rights Reserved/

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+    <modelVersion>4.0.0</modelVersion>
+    <packaging>war</packaging>
+    <version>1.0-SNAPSHOT</version>
+
+    <groupId>com.example.helloendpoints</groupId>
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml]: Is this used elsewhere?  It should be unique in the repo.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    <artifactId>helloendpoints</artifactId>
+
+    <properties>
+        <app.id>helloendpoints</app.id>
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml]: YOUR-APPID-HERE

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+        <app.version>1</app.version>
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml]: We don't usually supply this to the new plugins.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>display-dependency-updates</goal>
+                            <goal>display-plugin-updates</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml]: Not needed.  Default will work.

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+                <version>${maven.compiler.version}</version>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                  <source>${maven.compiler.source.version}</source>
+                  <target>${maven.compiler.target.version}</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml]: I suspect this isn't required either -- try w/o

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/src/main/java/com/example/helloendpoints/Constants.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/src/main/java/com/example/helloendpoints/Constants.java]: All Rights reserved (strike)

File: appengine/endpoints-v1-using-endpoints-frameworks-v2/src/main/java/com/example/helloendpoints/Greetings.java [hunk 14]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline appengine/endpoints-v1-using-endpoints-frameworks-v2/src/main/java/com/example/helloendpoints/Greetings.java]: ditto

**Comment by lesv** [comment]
Are you ready for me to look at this yet?

**Comment by frankyn** [comment]
@lesv PTAL

1. Acknowledging comments.
2. Adding a test to verify specific deployed version versus a prior deployment. 
3. Verified unique groupId for all repo.
4. Updated Copyrights
5. Updated README.md to reflect our conversation.

File: appengine/endpoints-frameworks-v2/README.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+This directory contains Google Cloud Endpoints Frameworks for App Engine for
+App Engine Standard samples. The [`backend/`](backend/) directory contains the
+sample code for the [quickstart][4] for Cloud Endpoints Frameworks on App Engine
+using an OpenAPI development process. The [`discovery/`](discovery/) directory
+contains the sample code for [migrated][2] prior version of [Cloud Endpoints
+Frameworks][1] project to the new [Cloud Endpoints Frameworks for App Engine][8]
+using a discovery document development process.
+
+The new Google Cloud Endpoints Frameworks for App Engine provides
+[additional functionality][3] using OpenAPI which is a paid service. It's
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/README.md]: which may require payment.

File: appengine/endpoints-frameworks-v2/discovery/Jenkins.sh [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+  curl -X GET \
+    "https://${1}.appspot.com/_ah/api/helloworld/v1/hellogreeting/1" | \
+    grep "goodbye world!"
+
+  # Test listGreeting Endpoint (hello world! and goodbye world!)
+  curl -X GET \
+    "https://${1}.appspot.com/_ah/api/helloworld/v1/hellogreeting" | \
+    grep "hello world!\|goodbye world!"
+
+  # Test multiply Endpoint (This is a greeting.)
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: This doesn't look like the right format to look at a specific instance.  `name-dot-instance.appspot.com` ??

File: appengine/endpoints-frameworks-v2/discovery/Jenkins.sh [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+# limitations under the License.
+
+# Fail on non-zero return and print command to stdout
+set -xe
+
+# Jenkins Test Script
+function TestEndpoints () {
+  # Test getGreeting Endpoint (hello world!)
+  curl -X GET \
+    "https://${1}.appspot.com/_ah/api/helloworld/v1/hellogreeting/0" | \
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: Shouldn't this be `{$1}`

File: appengine/endpoints-frameworks-v2/discovery/Jenkins.sh [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+# Modify Greetings.java for Gradle
+sed -i'.bak' -e "s/hello version-${GOOGLE_VERSION_ID}!/hello version-gradle-${GOOGLE_VERSION_ID}!/g" src/main/java/com/example/helloendpoints/Greetings.java
+
+# Deploy Gradle
+gradle appengineDeploy
+
+# End-2-End tests
+TestEndpoints $GOOGLE_PROJECT_ID "gradle-${GOOGLE_VERSION_ID}"
+
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: If you are going to use `${GOOGLE_VERSION_ID}` from Jenkins, you should delete your instances after you are done deploying.

File: appengine/endpoints-frameworks-v2/discovery/pom.xml [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+    <artifactId>helloendpoints</artifactId>
+
+    <properties>
+        <!-- Uncomment to use Endpoints Frameworks v1.0 -->
+        <!--
+        <appengine.endpoints.version>1.9.48</appengine.endpoints.version>
+        -->
+        <!-- End of Endpoints Frameworks v1.0 -->
+        <appengine.maven.version>1.0.0</appengine.maven.version>
+        <endpoints.framework.maven.version>1.0.0-beta2</endpoints.framework.maven.version>
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/pom.xml]: `<endpoints.maven.plugin.version>` ??

File: appengine/endpoints-frameworks-v2/discovery/pom.xml [hunk 20]
```diff
@@ -0,0 +1,124 @@
...
+            <version>${servlet.api.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+            <version>${javax.inject.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.jdo</groupId>
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/pom.xml]: I'll need to look at your code, but I'm a bit surprised your using this.  Isn't this SQL focused?

File: appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Constants.java [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+limitations under the License.
+*/
+
+package com.example.helloendpoints;
+
+/**
+ * Contains the client IDs and scopes for allowed clients consuming the helloworld API.
+ */
+public class Constants {
+  public static final String WEB_CLIENT_ID = "replace this with your web client ID";
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Constants.java]: Do you mention any of this in the README??

File: appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Greetings.java [hunk 22]
```diff
@@ -0,0 +1,84 @@
...
+import com.google.api.server.spi.config.ApiMethod;
+import com.google.api.server.spi.response.NotFoundException;
+import com.google.appengine.api.users.User;
+
+import java.util.ArrayList;
+
+import javax.inject.Named;
+//[END begin]
+ //[START api_def]
+
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Greetings.java]: Maybe the blank line should be between `END` & `START`  ??

File: appengine/endpoints-frameworks-v2/discovery/pom.xml (hunk 20 above)
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/pom.xml]: I don't see this being used anywhere, what's up???

File: appengine/endpoints-frameworks-v2/discovery/pom.xml [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+    <modelVersion>4.0.0</modelVersion>
+    <packaging>war</packaging>
+    <version>1.0-SNAPSHOT</version>
+
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/pom.xml]: Take a look at other appengine `pom.xml`, you should reference the parent `pom.xml`, then do a `mvn verify`

File: appengine/endpoints-frameworks-v2/README.md (hunk 15 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/README.md]: Acknowledged.

File: appengine/endpoints-frameworks-v2/discovery/Jenkins.sh (hunk 17 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: It should be ${1}.

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion

File: appengine/endpoints-frameworks-v2/discovery/Jenkins.sh (hunk 16 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: This didn't come to mind. Thanks for pointing that out!

File: appengine/endpoints-frameworks-v2/discovery/pom.xml (hunk 19 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/pom.xml]: Acknowledged.

File: appengine/endpoints-frameworks-v2/discovery/Jenkins.sh (hunk 18 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: Jenkins uses the GOOGLE_VERSION_ID as the test version and it expects to delete that version when tests are complete. Do you want this step to be contained within the Jenkins.sh?

File: appengine/endpoints-frameworks-v2/discovery/pom.xml (hunk 20 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/pom.xml]: Good catch. The sample doesn't use SQL whatsoever. I missed this dependency during clean up.

File: appengine/endpoints-frameworks-v2/discovery/pom.xml (hunk 23 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/pom.xml]: Acknowledged.

File: appengine/endpoints-frameworks-v2/discovery/Jenkins.sh (hunk 18 above)
Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: Yes - it hasn't been deleting things.  We ran out of versions last week.

Comment (lesv) [inline appengine/endpoints-frameworks-v2/discovery/Jenkins.sh]: Have you run this past the google bash linter?  And made sure you are with the style guide?  https://google.github.io/styleguide/shell.xml

File: appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Constants.java (hunk 21 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Constants.java]: No I did not. I updated the README with information about Google Accounts Client Ids.

File: appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Greetings.java (hunk 22 above)
Comment (frankyn) [inline appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Greetings.java]: Acknowledged

**Comment by frankyn** [comment]
@lesv PTAL

**Pull request merged**